ace_testbed/defiant_quick_start_guide.rst

```diff
@@ -0,0 +1,1142 @@
+``PrgEnv-<compiler>`` modules also defines a set of compiler wrappers for that
+compiler toolchain that automatically add include paths and link in libraries
+for Cray software. Compiler wrappers are provided for C (``cc``), C++ (``CC``),
+and Fortran (``ftn``).
+
+.. note::
+   Use the ``-craype-verbose`` flag to display the full include and link information
+   used by the Cray compiler wrappers. This must be called on a file to see the full
+   output (e.g., ``CC -craype-verbose test.cpp``).
+
+MPI
+---
+
+The MPI implementation available on Defiant is Cray's MPICH, which is "GPU-aware"
+so GPU buffers can be passed directly to MPI calls. Currently, Defiant has MPICH
+versions 8.1.27 as default and 8.1.26 as an additional module.
+
+----
+
+Compiling
+=========
+
+This section covers how to compile for different programming models using the
+different compilers covered in the previous section.
+
+MPI
+---
+
++----------------+----------------+-----------------------------------------------------+-------------------------------------------------------------------------------+
+| Implementation | Module         | Compiler                                            | Header Files & Linking                                                        | 
++================+================+=====================================================+===============================================================================+
+| Cray MPICH     | ``cray-mpich`` | ``cc``, ``CC``, ``ftn`` (Cray compiler wrappers)    | MPI header files and linking is built into the Cray compiler wrappers         |
+|                |                +-----------------------------------------------------+-------------------------------------------------------------------------------+
+|                |                | ``hipcc``                                           | | ``-L$(MPICH_DIR)/lib -lmpi``                                                |
+|                |                |                                                     | | ``-I$(MPICH_DIR)/include``                                                  |
++----------------+----------------+-----------------------------------------------------+-------------------------------------------------------------------------------+
+
+
+GPU-Aware MPI
+^^^^^^^^^^^^^
+
+To use GPU-aware Cray MPICH, there are currently some extra steps needed in
+addition to the table above, which depend on the compiler that is used.
+
+1. Compiling with the Cray compiler wrappers, ``cc`` or ``CC``
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+To use GPU-aware Cray MPICH with the Cray compiler wrappers, users must load
+specific modules, set some environment variables, and include appropriate headers
+and libraries. The following modules and environment variables must be set:
+
+.. note:: 
+
+    Setting ``MPICH_SMP_SINGLE_COPY_MODE=CMA`` is required as a temporary
```

Comment:
This is referencing a known issue that seems to be resolved. It might've been something that was present in spock. Should verify its still an issue.

Reply:
Issue is resolved and we are at MPICH version 8.1.27.